fix: toon shader + emission + transparencies visual artifacts

BrianAmadori opened the pull request:
This PR fixes the following:

- Fix: alpha expected to be multiplied by emission luminance on transparent meshes. (i.e. some fake lights were not looking good, this can be tested in Vegas plaza around the DJ field).

- Fix: Emission was not working with wearables.

- Fix: we had arithmetic exceptions turned on, this made the build crash in some cases.

- Fix: Avatar skin looking too dark in night mode

- Reduced avatar bloom multiplier from 5 to 4

- Fixed nullRef that could happen when wearables loading failed

- Fixed warning noise related to `CanvasRenderer` being present in our `TextShape` prefab

Also this PR introduces the `ShaderVariantTracker` tool. You can drag the prefab to the scene and it will modify the given `shaderVariantCollection` shader with any non-warmed shaders it will stumble upon. Note that this only can be done from Editor and aims to be useful in preventing shader runtime compilation.

Commit:
fix: toon shader artifacts

- added emission to shadergraph stack
- added our x5 emission multiplier to custom lighting override (to make bloom match our previous version)

Commit:
add shader variant tracker

Commit:
lit shader fixes

- multiply alpha by emission to fix existing transparency effects
- add variants detected by the variant tracker

Commit:
set arithmetic exceptions to ignore to avoid crashes

Commit:
fix: toon shader tweaks

- removed ambient lighting
- decreased emission multiplier to 4

Commit:
Merge branch 'master' into fix/unity-2020-visual-artifacts

nearnshaw commented:
FYI, trying this branch I see some transparent materials are not showing at all now

For example the glass domes or the spaceship glasses here:
https://play.decentraland.zone/branch/fix/unity-2020-visual-artifacts/index.html?ENV=org&position=52%2C4&realm=unicorn-blue

<img width="1111" alt="Screen Shot 2021-04-14 at 23 54 37" src="https://user-images.githubusercontent.com/3507907/114807592-109b0600-9d7d-11eb-889b-08a1e2913ad8.png">

D4rWiNSS commented on unity-client/Assets/Rendering/Shaders/Toon/Compiled/DCLSimpleLighting.hlsl:
```diff
@@ -37,16 +37,15 @@ half4 DCL_SimpleFragmentPBR(InputData inputData, SurfaceData surfaceData)
-
-    color += surfaceData.emission;
+    
+    const float DCL_EMISSION_MULTIPLIER = 4;
```
Is anyway we can set this constant outside the file without taking much time? It is used in several place and I am worried that if we change it in 1 shader, we forget to do it in all the others

pravusjif commented on unity-client/ProjectSettings/ProjectSettings.asset:
```diff
@@ -577,7 +577,7 @@ PlayerSettings:
   webGLUseEmbeddedResources: 0
   webGLCompressionFormat: 0
-  webGLWasmArithmeticExceptions: 0
+  webGLWasmArithmeticExceptions: 1
```
are you sure the arithmetic exceptions are being toggled off here? since I see it changes from 0 to 1

BrianAmadori replied:
0 == Throw
1 == Ignore

Not very intuitive but it works.
![image](https://user-images.githubusercontent.com/6875814/114917483-619e0f00-9dfc-11eb-8820-354ebb760b4f.png)

pravusjif reviewed (changes requested):
The code looks good, but we need to address the transparency issue detected by Nico.

**This branch build:** https://play.decentraland.zone/branch/fix/unity-2020-visual-artifacts/index.html?ENV=zone&position=65,-66

![Screen Shot 2021-04-15 at 09 33 01](https://user-images.githubusercontent.com/1031741/114870622-c478b180-9dce-11eb-8dbd-5159de99dbe6.png)

**Master branch build:** https://play.decentraland.zone/?ENV=zone&position=65,-66

![Screen Shot 2021-04-15 at 09 34 59](https://user-images.githubusercontent.com/1031741/114870730-e40fda00-9dce-11eb-8f4c-d69ca518a111.png)

Commit:
fix nullref

Commit:
fix: "remove canvas renderer component" warning

Commit:
fix: transparencies not working for meshes without emission maps

also, removed the upper bounds the emission alpha multiplication to work better with HDR

Commit:
refactor: moved our shader constants to DCL_CONSTANTS

Commit:
Merge branch 'master' into fix/unity-2020-visual-artifacts

Commit:
Merge branch 'master' into fix/unity-2020-visual-artifacts

pravusjif reviewed (approved):
bugs fixed, great!

Commit:
Merge branch 'master' into fix/unity-2020-visual-artifacts
